### PR TITLE
Workflow package for end-to-end testing

### PR DESCRIPTION
**Pull request by wild-endeavor**
This introduces a folder containing a small number of trivial workflows and tasks, emulating a user's Flyte repository, as a means for end-to-end testing.  End-to-end testing tests the Flyte platform as a whole, as a means to ensure that changes across multiple components work.

### CODE REVIEW

File: flytetester/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+# Install the AWS cli separately to prevent issues with boto being written over
+RUN pip3 install awscli
+
+# Virtual environment
+RUN python3.6 -m venv ${VENV}
+RUN ${VENV}/bin/pip install wheel
+
+# Install Python dependencies
+# TODO: Remove the pin of pip. In July-Aug 2019, it seemed like the latest version of pip had issues pulling from private
```
Comment (jonathanburns) [inline flytetester/Dockerfile]: Since it's no longer private, maybe you can remove?

File: flytetester/end2end/Readme.rst [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+#############
+Iteration
+#############
+
+For use with local Docker for Mac.  This is the easiest iteration cycle I've found in writing these end-to-end tests.
+
+#. Setup all the port-forwards for all the services (Admin, minio, & psql).  Just to make life easier.
+#. Truncate all tables in your local Admin Postgres database.
+#. Make changes to this repo.  Commit them, and build an updated Dockerhub image.
```
Comment (jonathanburns) [inline flytetester/end2end/Readme.rst]: Does this file still make sense for OSS? They can't ship dockerhub images from laptop.

File: flytetester/end2end/validator.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+    task_node = None
+    for n in ne:
+        if n.id.node_id == 'odd-nums-task':
+            task_node = n
+    assert task_node is not None
+
+    # Get the output from Minio into the workflow container
+    output_path = task_node.closure.output_uri
+    assert output_path != ''
+    tmp_file_name = '/tmp/output-{}'.format(execution.id.name)
```
Comment (jonathanburns) [inline flytetester/end2end/validator.py]: "mktemp" is better, but not a blocker, since this is just running in CI

Comment (wild-endeavor) [reply flytetester/end2end/validator.py]: will do this in the future.

File: flytetester/end2end/Readme.rst (hunk 2 above)
Comment (wild-endeavor) [reply flytetester/end2end/Readme.rst]: fixed.

File: flytetester/Dockerfile (hunk 1 above)
Comment (wild-endeavor) [reply flytetester/Dockerfile]: removed.

**Pull request merged**